### ISSUE TITLE
After update to Meteor 1.2 getting WARNING: Tried to load angular more than once, just for one template

### ISSUE THREAD

**Issue by EvilJohnny** [Urigo/angular-meteor#886]
Ok, I know this is weird, and I don't even know under what exact circumstances this is happening. I'll try to explain the best I can.

So  I have a module with a ui-router $state defined like this:

```javascript
angular.module('app.manager.events.newevent', ['app.shared'])

.config(
['$urlRouterProvider', '$stateProvider',
function($urlRouterProvider, $stateProvider){
	
	$stateProvider
		.state('manager.events.newevent', {
			url: '/newevent?name',
			templateUrl: 'client/manager/events/newevent/newevent.ng.html',
			controller: 'ManagerEventsNewEventCtrl'
		});
}]);
```

This was working perfectly ok, but after upgrading to Meteor 1.2, when going to this 'manager.events.newevent', I get the following errors:

```
WARNING: Tried to load angular more than once.
angular.js:12477 Error: only one instance of babel/polyfill is allowed
    at Object.eval (eval at <anonymous> (http://localhost:3000/packages/jquery.js?bd30605bc9f8429d01bae2d29081902d10a6c400:373:22), <anonymous>:2963:9)
    at Object.2.core-js/shim (eval at <anonymous> (http://localhost:3000/packages/jquery.js?bd30605bc9f8429d01bae2d29081902d10a6c400:373:22), <anonymous>:2966:4)
    at s (eval at <anonymous> (http://localhost:3000/packages/jquery.js?bd30605bc9f8429d01bae2d29081902d10a6c400:373:22), <anonymous>:2861:254)
    at e (eval at <anonymous> (http://localhost:3000/packages/jquery.js?bd30605bc9f8429d01bae2d29081902d10a6c400:373:22), <anonymous>:2861:425)
    at eval (eval at <anonymous> (http://localhost:3000/packages/jquery.js?bd30605bc9f8429d01bae2d29081902d10a6c400:373:22), <anonymous>:2861:443)
    at eval (eval at <anonymous> (http://localhost:3000/packages/jquery.js?bd30605bc9f8429d01bae2d29081902d10a6c400:373:22), <anonymous>:7124:4)
    at eval (eval at <anonymous> (http://localhost:3000/packages/jquery.js?bd30605bc9f8429d01bae2d29081902d10a6c400:373:22), <anonymous>:7131:3)
    at eval (native)
    at http://localhost:3000/packages/jquery.js?bd30605bc9f8429d01bae2d29081902d10a6c400:373:22
    at Function.jQuery.extend.globalEval (http://localhost:3000/packages/jquery.js?bd30605bc9f8429d01bae2d29081902d10a6c400:374:7) <div ui-view="" class="container-fluid tab-content ng-scope" data-ng-animate="1">
```

So for me, the ```WARNING: Tried to load angular more than once``` error usually is because a problem with the template, so I checked it and everything is ok (it was working before, so pretty sure the file and the folder were the same).

This is the structure of the folder:

```
client
 |-- manager
      |-- events
           |-- newevent
                |   newevent.ctrl.js
                |   newevent.ng.html
                |
                |-- lib
                     |   newevent.module.js
```

I already tried to change ".ng.html" with just ".html" (just for that file), but it didn't work.

Something I realized is that, when browsing to that state, the browser is requesting the file:
```
http://localhost:3000/client/manager/events/newevent/newevent.ng.html
```
Like it wasn't finding the template in the templates cache (I guess when meteor builds, all templates are loaded in the template cache).

As I say, I don't know when this is happening exactly. I have more modules with the exact same folder and file structure as "newevent", and they work fine. I tried to reproduce it creating a new one, and seems that depending on the naming on the folder, I can reproduce it or not... but I don't know how accurate is this.

Any ideas?

**Comment by MilosStanic** [comment]
I'm getting it too....

**Comment by MilosStanic** [comment]
Definitely a problem with not recognizing location of templates.
In my case, it was a strange problem. In one of the templates i had `<a href="/login">login</a>`, and when I changed `href` to `ui-sref="login"` it worked.

**Comment by thecodinghouse** [comment]
I am having the same issue. I have two applications. It is working for one but not for another.
I checked the templates cached in below folder structure for the app which is working but there are not template cache files for the app which is not working.
* .meteor
   * local
     * build
        * program
           * web.browser
              * app
                 * client
                    * myfolder
                      * template.file_name.js (auto generated file for template cache)

**Comment by jakerad1979** [comment]
Same here

**Comment by EvilJohnny** [comment]
Just commenting what I already said in #888 about template caching, that I think the problem is not there.
I created a new Meteor 1.2 fresh project, copied over the code and installed the needed packages, and de result was exactly the same.

I am wondering if there is some way of maintaining Meteor 1.1 way of parsing templates, while using the new angular package. Just as a workaround until we don't know more about this problem, and also to test if that fixes it, that will mean that the problem is on the new template parsing packages. I am trying to remove "angular-templates" from the project, but it gives me the error: "angular-templates is not in this project." Maybe because it is installed as a depencency and not listed in "packages"...

**Comment by sebakerckhof** [comment]
You guys should take a look in the javascript files to see what templates are being put in the template cache. You should see blocks of code similar to:

```js
(function(){

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                                                     //
// packages/packagename/path/to/template.filename.js                                 //
//                                                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
                                                                                                                       //
angular.module('angular-meteor').run(['$templateCache', function($templateCache) { 
$templateCache.put('/packages/packagename/path/to/filename.html',"...html...");
}]);
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

}).call(this);
```

Maybe the file does get processed and included, but with a wrong filename.
There might be more issues than only the caching issue for renamed files.

**Comment by ghiscoding** [comment]
I was wondering if I was the only one having this problem and found this open issue... As it's mentioned earlier, this seems to be happening since latest 1.2 version.

**Here is what worked for me:**
Instead of using `templateUrl: 'client/parties/views/parties-list.html'`, I completely remove the path and just kept the template name `parties-list.html` and it worked!

**Comment by EvilJohnny** [comment]
Hhmmm.. interesting @sebakerckhof 

In my case, the "template cache file", which is named **template.newevent.ng.js** it is in fact storing a different template! This is the content of the file:

```javascript
(function(){
  angular.module('angular-meteor').run(['$templateCache', function($templateCache) { 
    $templateCache.put('client/manager/events/editevent/editevent.ng.html',"<div class=\"row\"> <event-editor></event-editor> </div>");
  }]);
}).call(this);
```

So it is storing the template **client/manager/events/editevent/editevent.ng.html**, which of course is not the one I need :stuck_out_tongue:

**Comment by EvilJohnny** [comment]
OK, I think I know what the problem is! (At least in my case :stuck_out_tongue: )

So the two templates in conflict in my case **editevent.ng.html** and **newevent.ng.html** have something in common. Well, they have everything in common, except the name: the content of the file is the same. (This is because some TODOs and that the directive used there behaves different depending on the $stateParams).

Probably the compiler is somehow ignoring the **newevent.ng.html** template, as it already has one with the same content in cache (or something like that). I changed the content just adding some div, and everything worked.

I am not sure if this happens in other slightly similar cases... as I remember reproducing the issue multiple times with not the exact same folder structure and/or template content... But maybe in thoses cases the cache had something to do...

What I am totally sure, is that after changing on of the templates a bit, and removing cached files, it definitely worked.

**Comment by sebakerckhof** [comment]
@EvilJohnny Then it's the same issue as: https://github.com/Urigo/angular-meteor/issues/888

It's because the caching only looks at the source code to generate the cache key. Should be fixed with this PR: https://github.com/Urigo/angular-meteor/pull/889

**Comment by EvilJohnny** [comment]
@sebakerckhof You are right!

Sounds good to me :smiley:

**Comment by EvilJohnny** [comment]
Fixed in 1.2.3
